src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,188 @@
+import {
+  ChangeEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { getSearchWith } from '../../utils/searchHelper';
+import { SortOrder } from '../../utils/SortOrder';
+import { Loader } from '../Loader';
+import { PeopleFilters } from '../PeopleFilters';
+import { PeopleTable } from '../PeopleTable';
+
+const filterPeople = (
+  people: Person[],
+  sex: string,
+  query: string,
+  centuries: string[],
+): Person[] => {
+  let newPeople = [...people];
```

Comment:
Method `filter` doesn't change the existing array but returns the result in a brand new array. So you don't need the spread operator here.